portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -0,0 +1,40 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class HomeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/index.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      
+      User userData = new User(true, userEmail, logoutUrl);
+
+      response.setContentType("application/json;");
+      Gson gson = new Gson();
+      response.getWriter().println(gson.toJson(userData));
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/index.html";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      
+      User userData = new User(false, null, loginUrl);
+
+      response.setContentType("application/json;");
+      Gson gson = new Gson();
+      response.getWriter().println(gson.toJson(userData));
```

Comment:
you have quite a bit of duplicate code between these two cases - it seems like the only difference is the params in new User(). can you restructure this a bit to avoid the duplication?

Reply:
Agreed. Done